pkg/traces/servicegraphprocessor/processor.go

```diff
@@ -260,27 +273,28 @@ func (p *processor) consume(trace pdata.Traces) error {
 				case pdata.SpanKindClient:
 					k := key(span.TraceID().HexString(), span.SpanID().HexString())
 
-					var r edgeRequest
-					if v, ok := p.store.Get(k); ok {
-						r = v.(edgeRequest)
+					edge := p.store.upsertEdge(k, func(e *edge) {
+						e.clientService = svc.StringVal()
+						e.clientLatency = spanDuration(span)
+						e.failed = p.spanFailed(span)
```

Comment:
failed can be overwritten if one side shows success and one side shows failure. edge failure or success would then be determine by whichever span it randomly processed last.

if failed is true we should never change it back to false